src/algebra/lattice_ordered_group.lean

```diff
@@ -0,0 +1,607 @@
+
+This file develops the basic theory, concentrating on the commutative case written additively.
+
+## Main definitions
+
+* `lattice_ordered_group` - a (non-commutative) partially ordered (multiplicative) group which forms
+  a lattice with respect to the partial order.
+* `lattice_ordered_add_group` - the additive form of a `lattice_ordered_group`
+* `lattice_ordered_comm_group` - a lattice ordered (multiplicative) group with commutative group
+  operation
+* `lattice_ordered_add_comm_group` - the additice form of a `lattice_ordered_comm_group`
+
+## Main statements
+
+- `pos_sub_neg`: Every element `a` of a `lattice_ordered_add_comm_group` has a decomposition `a⁺-a⁻`
+   into the difference of the positive and negative component.
+- `pos_meet_neg_eq_zero`: The positive and negative components are coprime.
+- `abs_triangle`: The absolute value operation satisfies the triangle inequality.
+
+It is shown that the meet and join operations are related to the absolute value operation by a
+number of equations and inequalities.
+
+## Notations
+
+- `a⁺ = a ⊔ 0`: The *positive component* of an element `a` of a `lattice_ordered_add_comm_group`
+- `a⁻ = (-a) ⊔ 0`: The *negative component* of an element `a` of a `lattice_ordered_add_comm_group`
+* `|a| = a⊔(-a)`: The *absolute value* of an element `a` of a `lattice_ordered_add_comm_group`
+
+## Implementation notes
+
+We define `lattice_ordered_group` as extending `group` and `lattice` and
+`lattice_ordered_comm_group` as extending `comm_group` and `lattice`. Equivalent additive classes
+are defined. We then show that a `lattice_ordered_comm_group` is a `lattice_ordered_group` and an
+`ordered_comm_group`.
+
+The remainder of the file establishes basic properties of `lattice_ordered_add_comm_group`. A number
+of these results also hold in the non-commutative case ([Birkhoff][birkhoff1942],
+[Fuchs][fuchs1963]) but we have not developed that here. We have concentrated on groups written
+additively, since we are primarily interested in vector lattices.
+
+## References
+
+* [Birkhoff, Lattice-ordered Groups][birkhoff1942]
+* [Bourbaki, Algebra II][bourbaki1981]
+* [Fuchs, Partially Ordered Algebraic Systems][fuchs1963]
+* [Zaanen, Lectures on "Riesz Spaces"][zaanen1966]
+* [Banasiak, Banach Lattices in Applications][banasiak]
+
+## Tags
+
+lattice, ordered, group
+-/
+
+universe u
+
+/--
+A group `(α,*)`, equipped with a partial order `≤`, making `α` into a lattice, such that, given
+elements `a` and `b` of type `α` satisfying `a ≤ b`, for all elements `c` and `d` of `type α`,
+$$c * a * d ≤ c * b * d,$$
+is said to be a *lattice ordered group*.
+-/
+class lattice_ordered_group (α : Type u)
+  extends group α, lattice α :=
+(mul_le_mul : ∀ a b : α, a ≤ b → ∀ c d : α, c * a * d ≤ c * b * d)
+
+/--
+A lattice ordered group with Abelian group multiplication is said to be a *lattice ordered
+commutative group*.
+-/
+class lattice_ordered_comm_group (α : Type u)
```

Comment:
As above, instead of writing
```lean
[lattice_ordered_comm_group]
```
you should probably write
```lean
[lattice α] [comm_group α] [covariant_class α α (*) (≤)] [covariant_class α α (function.swap (*)) (≤)]
```
Let's keep the discussion to the above comment though